src/ui/pages/AssetInfo/AssetResources.tsx

```diff
@@ -0,0 +1,365 @@
+    <Button
+      key={resource.name}
+      as={UnstyledAnchor}
+      rel="noopenner noreferrer"
+      target="_blank"
+      kind="regular"
+      size={36}
+      href={resource.url}
+      style={{
+        padding: 6,
+        border: '2px solid var(--neutral-200)',
+        ['--button-background' as string]: 'var(--white)',
+        ['--button-background-hover' as string]: 'var(--neutral-200)',
+      }}
+      aria-label={resource.displayableName}
+    >
+      {icon}
+    </Button>
+  );
+}
+
+const TWITTER_ID = 'twitter';
+const WARPCAST_ID = 'warpcast';
+const DEXSCREENER_ID = 'dexscreener';
+
+export function AssetResources({
+  assetFullInfo,
+}: {
+  assetFullInfo: AssetFullInfo;
+}) {
+  const { networks } = useNetworks();
+  const dialogRef = useRef<HTMLDialogElementInterface | null>(null);
+  const implementations = useMemo(
+    () =>
+      Object.entries(assetFullInfo.fungible.implementations)
+        .map(([networkId, config]) => {
+          const network = networks?.getNetworkByName(createChain(networkId));
+          return network ? { address: config.address, network } : null;
+        })
+        .filter(isTruthy),
+    [assetFullInfo.fungible.implementations, networks]
+  );
+
+  const mainImplementation = useMemo(() => {
+    return (
+      implementations.find(
+        (implementation) =>
+          implementation.network.id === assetFullInfo.extra.mainChain
+      ) || implementations[0]
+    );
+  }, [implementations, assetFullInfo.extra.mainChain]);
+
+  const resourcesById = useMemo(() => {
+    return assetFullInfo.extra.relevantResources.reduce((acc, resource) => {
+      acc[resource.name] = resource;
+      return acc;
+    }, {} as Record<string, AssetResource>);
+  }, [assetFullInfo.extra.relevantResources]);
+
+  return (
+    <>
+      <HStack gap={8} justifyContent="space-between">
+        <HStack gap={8}>
+          {resourcesById[TWITTER_ID] ? (
+            <ResourseButton
+              resource={resourcesById[TWITTER_ID]}
+              icon={<XIcon style={{ width: 20, height: 20 }} />}
+            />
+          ) : null}
+          {resourcesById[WARPCAST_ID] ? (
+            <ResourseButton
+              resource={resourcesById[WARPCAST_ID]}
+              icon={<WarpcastIcon style={{ width: 20, height: 20 }} />}
+            />
+          ) : null}
+          <ResourseButton
+            resource={
+              resourcesById[DEXSCREENER_ID] || {
+                name: DEXSCREENER_ID,
+                displayableName: 'Dexscreener',
+                iconUrl: '',
+                url: `https://dexscreener.com/search?q=${assetFullInfo.fungible.symbol}`,
+              }
+            }
+            icon={<DexscreenerIcon style={{ width: 20, height: 20 }} />}
+          />
+        </HStack>
+        <HStack gap={4}>
+          {mainImplementation ? (
+            <MainImplementationButton implementation={mainImplementation} />
+          ) : null}
+          {implementations.length ? (
```

Comment:
maybe you meant this?
```suggestion
          {implementations > 1 ? (
```
It seems that `implementations` can't be empty if `mainImplementation` is defined

Reply:
This one is a thing to be improved in the future. Right now I show this button even if length = 1, cause in the dialog there is also an option to open the link in explorer, not only copy the address on chain

Reply:
Makes sense!